validmind/vm_models/dataset.py

```diff
@@ -241,18 +244,19 @@ def assign_predictions(
                 logger.info(
                     f"Prediction column {pred_column} already exist in the dataset. Linking the model with the {pred_column} column"
                 )
-            else:
-                logger.info("Running predict()... This may take a while")
-                x_only = self._raw_dataset[
-                    :,
-                    [
-                        self._columns.index(name)
-                        for name in self._columns
-                        if name in self._feature_columns
-                    ],
-                ]
-                prediction_values = np.array(model.predict(x_only))
-                self.__assign_prediction_values(model, pred_column, prediction_values)
+                return
+
+            logger.info("Running predict()... This may take a while")
+
+            # If the model is a FoundationModel, we need to pass the DataFrame to
+            # the predict method since it requires column names in order to format
+            # the input prompt with its template variables
+            x_only = (
+                self.x_df() if model.model_library() == "FoundationModel" else self.x
```

Comment:
@AnilSorathiya this was the issue with the foundation model tests: since they require column names to format prompts we're now passing a dataframe when the model is `FoundationModel`. Also, note that I simplified this code by using `self.x` which is already available.